MIPs/mip-x-multichainapi.md

```diff
@@ -0,0 +1,241 @@
+
+```js
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "wallet_createSession",
+  "params": {
+    "optionalScopes": {
+      "wallet:eip155": {
+        "methods": ["wallet_addEthereumChain"],
+      },
+      "eip155:1": {
+        "methods": ["eth_sendTransaction","eth_call","eth_getBalance","eth_blockNumber","eth_getTransactionCount","wallet_watchAsset","eth_subscribe"],
+        "notifications": ["eth_subscription"],
+      },
+      "solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp": {
+        "methods": ["signAndSendTransaction","signAllTransactions","signMessage"],
+        "notifications": ["accountChanged"],
+      }
+      "bip122:000000000019d6689c085ae165831e93": {
+        ...
+      }
+    },
+    "scopedProperties": {
+      // Scope-keyed proposals or declarations
+      "eip155:1": {
+        "extension_foo": "bar"    
+      }
+    },
+  },
+}
+```
+
+### Response
+An example structure for the corresponding JSON-RPC response that a dapp would receive from the wallet:
+
+```js
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "sessionScopes": { 
+      "eip155:wallet": {
+          "methods": ["wallet_addEthereumChain"],
+          "accounts": ["eip155:wallet:0x0910e12C68d02B561a34569E1367c9AAb42bd810"]
+      },
+      "eip155:1": {
+        "methods": ["eth_sendTransaction","eth_call","eth_getBalance","eth_blockNumber","eth_getTransactionCount","wallet_watchAsset","eth_subscribe"],
+        "notifications": ["eth_subscription"],
+        "accounts": ["eip155:1:0x0910e12C68d02B561a34569E1367c9AAb42bd810"]
+      },
+      "solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp": {
+        "methods": ["signAndSendTransaction","signAllTransactions","signMessage"],
+        "notifications": ["accountChanged"],
+        "accounts": ["solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp:7S3P4HxJpyyigGzodYwHtCxZyUQe9JiBMHyRWXArAaKv"]
+      }
+    },      
+  }
+}
+```
+
+# Proposal
+
+## Language
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written in uppercase in this document are to be interpreted as described in RFC 2119.
+
+## Definitions
+
+**Multichain API**: a new [Wallet API](https://docs.metamask.io/wallet/concepts/wallet-api/) that is generalized for interactions between dapps and wallets in a multichain environment
+
+**network**: Decentralized networks where control, validation, and decision-making processes are distributed across multiple nodes rather than centralized in a single entity. The term will refer to a specific decentralized network that has a single unique identifier. Networks may be implemented as a blockchain or another distributed ledger technology. 
+
+**node**: An RPC node that facilitates interactions with a decentralized network.
+
+**Wallet API**: [The JSON-RPC interface](https://docs.metamask.io/wallet/concepts/apis/) that dapps can use to programmatically interact with MetaMask wallet clients.
```

Comment:
Do you want to differentiate between the Wallet API's provider API and JSON-RPC API? If so, is the Multichain API more equivalent to the provider API?

Reply:
I'm not sure the Multichain API precisely maps to the current JSON-RPC API, but many concepts around the provider API (such as EIP-1193 patterns) are going away with the introduction of the Multichain API. The new API is less reliant on a JavaScript object interface and primarily involves JSON-RPC request/response.